resources/assets/js/components/ProgressTracker/ProgressTrackerItem.tsx

```diff
@@ -37,10 +49,35 @@ const ProgressTrackerItem: React.FunctionComponent<
           {state === "active" && <i className="fas fa-arrow-down" />}
         </div>
       )}
-      <div className="tracker-title">
-        <span data-c-font-size="small">{label}</span>
-        <span data-c-font-weight="bold">{title}</span>
-      </div>
+      {state !== "null" ? (
+        <a
+          href={link}
+          title={title}
+          onClick={(e): void => {
+            e.preventDefault();
+            if (
+              window.confirm(
+                intl.formatMessage(jobBuilderMessages.unsavedChangesWarning),
+              )
+            ) {
+              navigate(link);
+            }
+          }}
```

Comment:
Don't worry about throwing that alert here. We only want to show the alert when there is unsaved data, which means the form components themselves will have to intercept the navigation somehow.

p.s. We're trying to keep this component generic. If we kept this alert here, the message should be moved to somewhere other than jobBuilderMessages.

Reply:
I removed the onClick handler. We can make a better solution while working on #1850.